### PR TITLE
fix: added `usingCn` flag to `ComponentPreview`

### PR DESCRIPTION
**Pull request by epoll31**
## Description

I implemented an optional `usingCn` prop on `ComponentPreview` that will show step in how to add the `cn` function to a project.

## Proposed Changes

- renamed `lib/utils.ts` to `utils/cn.ts` which is more clear about what the file is for.
  -  this includes renaming all imports that reference `lib/utils`
- added optional `noExpand` prop that hides the expand/minimize functionality on `CodeGroup`
  - this is so that I can use `CodeGroup` with smaller sections where the minimize adds clutter instead of functionality
- removed simple `cn` implementation in `SimpleToggle.tsx` in favor of actual `cn` function
- updated simple-toggle docs page to use the new `usingCn` flag 

## Screenshots

Screenshot 2024-05-22 at 10 24 03 AM

## Checklist

Please check the boxes that apply:

- [x] I have rebased my branch on top of the latest `main` branch.
- [x] I have tested the changes locally
- [x] I ran `npm run build` and build is successful
- [x] I have added necessary documentation (if applicable)
- [x] I have updated the credits.md file (if applicable)

### CODE REVIEW

**Comment by epoll31** [comment]
Screenshot 2024-05-22 at 11 45 05 AM

Added steps and now only when `usingCn` is flagged the preview and code will render together instead of two tabs.

File: src/utils/cn.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { type ClassValue, clsx } from 'clsx'
```
Comment (Ansub) [inline src/utils/cn.ts:1]: make this path `/src/lib/utils.ts`

File: components.json [hunk 2]
```diff
@@ -12,6 +12,6 @@
   },
   "aliases": {
     "components": "@/components",
-    "utils": "@/lib/utils"
+    "utils": "@/utils/cn"
```
Comment (Ansub) [inline components.json:15]: fix this too, this will break shadcn/ui

File: src/lib/utils.ts
Comment (Ansub) [inline src/lib/utils.ts:1]: add this file back

File: src/components/ui/button.tsx [hunk 3]
```diff
@@ -1,36 +1,38 @@
-import * as React from "react"
```
Comment (Ansub) [inline src/components/ui/button.tsx:1]: revert this back

File: src/components/icons/CopyButton.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import { CheckIcon, ClipboardIcon } from 'lucide-react'
 import * as React from 'react'
 import { Event, trackEvent } from '../../../types/events'
-import { cn } from '@/lib/utils'
+import cn from '@/utils/cn'
```
Comment (Ansub) [inline src/components/icons/CopyButton.tsx:6]: revert this back

File: src/components/code/CodeGroup.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 'use client'
 
-import { cn } from '@/lib/utils'
```
Comment (Ansub) [inline src/components/code/CodeGroup.tsx:3]: revert this back

File: src/components/code/Code.tsx [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 import { useContext, useEffect, useMemo, useState } from 'react'
 import { CodeGroupContext } from './CodeGroup'
-import { cn } from '@/lib/utils'
```
Comment (Ansub) [inline src/components/code/Code.tsx:5]: revert this back

**Pull request merged**